framework/source/class/qx/ui/window/Window.js

```diff
@@ -993,6 +1018,112 @@ qx.Class.define("qx.ui.window.Window",
       }
     },
 
+    // overridden
+    _transformCenterWhen : function(value)
+    {
+      var             values = value;
+      
+      // Validate type of value
+      if (typeof value != "string" && ! (value instanceof Array))
+      {
+        throw new Error(
+          "centerWhen requires a string or an array of strings");
+      }
+
+      // Convert the value to an array if it was given as a string.
+      if (typeof value == "string")
+      {
+        if (value.trim() === "")
+        {
+          // We were given an empty string or array, requesting no automatic
+          // centering. Create an empty array of centering behaviors.
+          values = [];
+        }
+        else
+        {
+          // We were given a string. If they put multiple values into the
+          // string, separate them out into separate values, and place all into
+          // an array.
+          values = qx.lang.String.clean(value).split(/\s+/);
+        }
+      }
+
+      // Validate
+      values.forEach(
+        function(value)
+        {
+          switch(value)
+          {
+          case "appear" :
+          case "resize" :
+            // These are expected values
+            break;
+
+          default :
+            throw new Error("Unexpected centerWhen value: " + value);
+          }
+        });
+
+      return values;
+    },
```

Comment:
I see your point - but is it necessary to convert from strings in various forms?  The centerWhen property has a given type (array) and set of allowed values in that array, I don't think it's appropriate to coerce the values from a subset of other unknown values, especially if that means parsing strings into arrays etc.  IMHO it's the caller's requirement to use the correct type.

As a side issue, I agree with the usefulness of sets as property values, it would be great if there was a way to express this in qx property definitions but perhaps we can talk about that on gitter?